src/gui/wizard/owncloudwizard.cpp

```diff
@@ -152,6 +188,19 @@ int OwncloudWizard::calculateLongestSideOfWizardPages(const QList<QSize> &pageSi
     });
 }
 
+void OwncloudWizard::reject()
+{
+    QWizard::reject();
+
+    // Hide menubar and dock icon
+    CocoaProcessTypeToBackground();
```

Comment:
If you remember I was complaining about that approach earlier and advocated for event filter. This is a sign of why I was advocating for that. Now the two PRs are interwined but with the event filter approach they can be cleanly separated. There's better code decoupling.

Could you remove all the foreground/background related stuff from that PR? This would make it easier to review as well.

Reply:
I hadn't gotten around to updating this PR to match the foreground/background logic in it. FWIW the wizard window is already maximizable on macOS, and it suffers from the problem of not being able to be unmaximized because it doesn't have the default menubar. Like, literally, if you mouse up to the top of the screen, it doesn't show the titlebar of the window and consequently doesn't show the "stoplight" buttons. The only way to unmaximize the window is to open Launchpad (i.e. the Activity Overview), mouse over the maximized window at the top of the screen, wait for a short delay before the unmaximize button appears on top of it there, then click that.

That said, the issue with maximizing and unmaximizing the wizard window is really tangential to the core issue here of not being able to quit Nextcloud Desktop with no sync accounts set up. By contrast, the changes in #2959 consist almost entirely of the background/foreground logic. It would make more sense to have a PR for background/foreground logic precede a PR for being able to maximize the settings dialog, specifically in order to avoid porting the wizard's problem of not being able to unmaximize to the settings dialog, as well.

Reply:
My thought is that it would make the most sense to have three different pull requests:

1. Allow Quit from Initial Setup
2. Show/Hide Menubar and Dock Icon for Persistent Windows (Prerequisite for item (3))
3. Allow Maximizing Settings Window

The main question would probably be whether #2959 should become item (2) or item (3). I'll try and make up my mind in the next or so, idk.

Reply:
Since most of the conversation on #2959 has to do with the foreground/background logic, not the `Qt::Window` flag, I'm inclined to have the `Qt::Window` flag be the part that moves to a new and new pull request (and I can rename the existing branch and existing pull request after I do so).

Reply:
Apparently I can't rename this branch without it closing this pull request.